code/datums/action.dm

```diff
@@ -673,23 +681,71 @@
 /datum/action/cooldown/IsAvailable()
 	return next_use_time <= world.time
 
-/datum/action/cooldown/proc/StartCooldown()
-	next_use_time = world.time + cooldown_time
-	button.maptext = MAPTEXT("<b>[round(cooldown_time/10, 0.1)]</b>")
+/// Starts a cooldown time, will use default cooldown time if an override is not specified, shares that cooldown with similar abilities
+/datum/action/cooldown/proc/StartCooldown(override_cooldown_time, will_share_cooldowns = TRUE)
```

Comment:
The boolean parameter shouldn't be here, this should just be two separate procs, with one being the process after this that starts the cooldown, and nothing else.

The two places in which you actually use `will_share_cooldowns = FALSE` you would call that directly. `StartCooldown()` itself would run the shared_cooldown stuff, though.